Changes for kfserving 0.4.1

juliusvonkohout opened the pull request:
**Description of your changes:**


**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [x] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

juliusvonkohout commented:
@googlebot I signed it!

animeshsingh reviewed (commented):
/lgtm

cc @Tomcli

juliusvonkohout commented:
> Thanks @juliusvonkohout, can you also update this line to version 0.4.0 as well? I updated the 0.4.0 image to based on your branch.
> 
> https://github.com/kubeflow/pipelines/blob/41dc3865eb053bff4eb77029cd56a7bed2b46f0e/components/kubeflow/kfserving/component.yaml#L21
> 
> Otherwise it looks good to me.

I have changed the version now.

Tomcli commented:
/lgtm

Tomcli commented:
/ok-to-test

DavidSpek commented:
@juliusvonkohout Does this need a refacotr for kfserving 0.4.1?

juliusvonkohout commented:
@DavidSpek tehere are two places where 0.4.0 should be replaced with 0.4.1 and it should be rebased against master

juliusvonkohout commented:
> Thanks @juliusvonkohout, can you also update this line to version 0.4.0 as well? I updated the 0.4.0 image to based on your branch.
> 
> https://github.com/kubeflow/pipelines/blob/41dc3865eb053bff4eb77029cd56a7bed2b46f0e/components/kubeflow/kfserving/component.yaml#L21
> 
> Otherwise it looks good to me.

i guess you have to update to 0.4.1 and test again. Feel free to make any edits.

Tomcli commented on components/kubeflow/kfserving/src/kfservingdeployer.py:
```diff
@@ -407,12 +407,21 @@ def update(kfsvc, model_name, namespace):
         print("Sample test commands: \n")
         # model_status['status']['url'] is like http://flowers-sample.kubeflow.example.com/v1/models/flowers-sample
+
+        host, path = url.sub("", model_status["status"]["address"]["url"]).split("/", 1)
```
Few weeks ago, we discovered that this split command may not work with some DNS for OpenShift route. So we removed it to avoid any confusion and error.

Tomcli commented:
@juliusvonkohout the kubeflow-pipelines-component-yaml test is fixed now. Can you rebase this PR and remove the old changes from line 410-422? thanks

juliusvonkohout commented:
@tomcli I strongly hope that my resolve was correct. Someone should test this first.

Tomcli commented on components/kubeflow/kfserving/src/kfservingdeployer.py:
```diff
@@ -429,12 +429,21 @@ def update(kfsvc, model_name, namespace):
+        host, path = url.sub("", model_status["status"]["address"]["url"]).split("/", 1)
         print(
-            "curl -v -X GET %s" % model_status["status"]["url"]
+            'curl -X GET -H "Host: ' + host + '" http://$ISTIO_INGRESS_ENDPOINT/' + path
```
```suggestion
            "curl -v -X GET %s" % model_status["status"]["address"]["url"]
```

Tomcli commented on components/kubeflow/kfserving/src/kfservingdeployer.py:
```diff
@@ -429,12 +429,21 @@ def update(kfsvc, model_name, namespace):
-            model_status["status"]["url"] + " is the knative domain."
+            model_status["status"]["address"]["url"] + " is the knative domain."
+        )
+        print("Sample test commands: ")
```
can you remove line 434-439, and line 444? thanks

juliusvonkohout replied:
as you wish

Tomcli reviewed (changes requested):
@juliusvonkohout I have few comments for the rebase, thanks.

juliusvonkohout commented:
We should squash commit such that there are not so many garbage mini edits from me.

Tomcli commented:
/lgtm
I added the 0.4.1 image tag for this component.yaml

Tomcli commented:
@animeshsingh can you approve this PR? Thanks.

animeshsingh commented:
/approve

juliusvonkohout commented:
@tomcli thank you very much for this effort. I already started to think that it will just be forgotten or is not compatible anymore.
